src/goose.rs

```diff
@@ -1127,8 +1130,12 @@ impl GooseUser {
             }
         }
 
-        // Otherwise use the `base_url`.
-        Ok(self.base_url.join(path)?.to_string())
+        // Otherwise use the `base_url` after checking that it isn't unspecified.
+        Ok(self
+            .base_url
+            .as_ref()
+            .ok_or(url::ParseError::EmptyHost)?
```

Comment:
Using `EmptyHost` is misleading as we are not trying to parse a string as a URL, I think returning an explicit error with a custom message would be better. Something like `cannot create single user, unspecified base_url`.

Reply:
Sg 👍🏼 Should I create a new `TransactionError` enum value for this?

Reply:
After understanding the code better makes sense to leave it as a UrlParse error since the TransactionError doesn't have a generic error variant.